src/dotnet-svcutil/lib/src/CommandProcessorOptions.cs

```diff
@@ -0,0 +1,1059 @@
+                if (cmdOptions.Errors.Count() == 0 && cmdOptions.Inputs.Count == 1)
+                {
+                    if (PathHelper.IsFile(cmdOptions.Inputs[0], Directory.GetCurrentDirectory(), out var fileUri) &&
+                        TryFromFile(fileUri.LocalPath, out var fileOptions) && fileOptions.GetOptions().Count() > 0)
+                    {
+                        // user switches are disallowed when a params file is provided.
+                        var options = cmdOptions.GetOptions().ToList();
+                        var disallowedSwitchesOnParamsFilesProvided = CommandSwitch.All
+                            .Where(s => !CmdLineOverwriteSwitches.Contains(s.Name) && s.SwitchLevel <= OperationalContext.Global && options.Any(o =>
+                            {
+                                if (o.HasSameId(s.Name))
+                                {
+                                    o.Value = null;
+                                    return true;
+                                }
+                                return false;
+                            }));
+
+                        // warn about disallowed options when params file has been provided and clear them.
+                        if (disallowedSwitchesOnParamsFilesProvided.Count() > 0)
+                        {
+                            fileOptions.AddWarning(string.Format(SR.WrnExtraParamsOnInputFileParamIgnoredFormat, disallowedSwitchesOnParamsFilesProvided.Select(s => $"'{s.Name}'").Aggregate((msg, n) => $"{msg}, '{n}'")), 0);
+                        }
+
+                        fileOptions.ResolveFullPathsFrom(new DirectoryInfo(Path.GetDirectoryName(fileUri.LocalPath)));
+
+                        // ensure inputs are clear as the params file is the input.
+                        cmdOptions.Inputs.Clear();
+                        cmdOptions.CopyTo(fileOptions);
+                        cmdOptions = fileOptions;
+                    }
+                }
+            }
+            catch (ArgumentException ae)
+            {
+                cmdOptions.AddError(ae);
+            }
+            catch (FileLoadException fle)
+            {
+                cmdOptions.AddError(fle);
+            }
+            catch (FormatException fe)
+            {
+                cmdOptions.AddError(fe);
+            }
+
+            await cmdOptions.ProcessBasicOptionsAsync(logger, cancellationToken);
+
+            return cmdOptions;
+        }
+
+        public async Task ResolveAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (this.Help != true && this.Errors.Count() == 0)
+                {
+                    ProcessLanguageOption();
+
+                    ProcessSerializerOption();
+
+                    // process project file first as it can define the working directory.
+                    await ProcessProjectFileOptionAsync(cancellationToken).ConfigureAwait(false);
+
```

Comment:
ConfigureAwait is only needed when either exposed as a publicly consumable api or when running in a UI app.

Reply:
This is part of the library which can be called from the WCF Connected Service, which is a UI app.